quinn-proto/src/connection/mod.rs

```diff
@@ -851,7 +854,14 @@ impl Connection {
             &self.config,
         );
 
-        let pto = self.ack_frequency.max_ack_delay_for_pto() + data.rtt.pto_base();
+        let pto = match rtt_hint {
+            Some(rtt_hint) => rtt_hint,
+            None => {
+                // Fallback to the pessimistic calculation
+                // TODO: we could consider using known PTOs in this case
+                self.ack_frequency.max_ack_delay_for_pto() + data.rtt.pto_base()
+            }
+        };
```

Comment:
```suggestion
        if let Some(rtt_hint) = rtt_hint {
            data.rtt = RttEstimator::new(rtt_hint);
        }
        let pto = match rtt_hint {
            Some(rtt_hint) => rtt_hint,
            None => {
                // Fallback to the pessimistic calculation
                // TODO: we could consider using known PTOs in this case
                self.ack_frequency.max_ack_delay_for_pto() + data.rtt.pto_base()
            }
        };
        let pto = self.ack_frequency.max_ack_delay_for_pto() + data.rtt.pto_base();
```

That seems more correct to me...
We don't want to only change the PathOpen timer, we also want to change the time at which we resend path challenges, and change when we consider packets lost, etc.
Also, the PTO would actually be a different value in this case, because we make assumptions about variance, and add the max ACK delay. It surprises me that all of that would be skipped in your version.